### PR TITLE
Item repo brain refactor

### PR DESCRIPTION
**Pull request by Sidarth20**
# Description

Updated spacing, ensured every method had a test, tested other side of conditional.

# Been Tested?

 - [x]

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] !!!!!Every method has a test!!!!
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Require 'simplecov' 
- [x] SimpleCov.start
- [x] Used require_relative

### CODE REVIEW

File: spec/item_repository_spec.rb [hunk 1]
```diff
@@ -287,4 +325,38 @@
...
+      ir = ItemRepository.new('spec/truncated_data/items_truncated.csv', se)
+
+      expect(ir.merchants_with_only_one_item.count).to eq(3)
+      expect(ir.merchants_with_only_one_item[1].name).to eq("MiniatureBikez")
+    end
+  end
+
+  describe '#items_created_in_month' do
+    it 'creates a hash of number of items created in given month ' do
+      se = SalesEngine.from_csv({
```
Comment (Sidarth20) [inline spec/item_repository_spec.rb]: removed

File: spec/item_repository_spec.rb [hunk 2]
```diff
@@ -287,4 +325,38 @@
...
+          merchants: './spec/truncated_data/merchants_truncated.csv',
+          invoices: './spec/truncated_data/invoices_truncated.csv',
+          customers: './spec/truncated_data/customers_truncated.csv',
+          invoice_items: './spec/truncated_data/invoice_items_truncated.csv',
+          transactions: './spec/truncated_data/transactions_truncated.csv'
+                                })
+      ir = ItemRepository.new('spec/truncated_data/items_truncated.csv', se)
+
+      expect(ir.merchants_with_only_one_item.count).to eq(3)
+      expect(ir.merchants_with_only_one_item[1].name).to eq("MiniatureBikez")
```
Comment (Sidarth20) [inline spec/item_repository_spec.rb]: fixed.

File: spec/item_repository_spec.rb [hunk 3]
```diff
@@ -287,4 +325,38 @@
       expect(ir.golden_items).to eq([])
     end
   end
+
+  describe '#merchants_with_only_one_item' do
+    it 'finds merchants with only 1 item and returns item object' do
+      se = SalesEngine.from_csv({
```
Comment (Sidarth20) [inline spec/item_repository_spec.rb]: removed

File: spec/item_repository_spec.rb [hunk 4]
```diff
@@ -177,12 +206,18 @@
...
+
       expect(test_item.name).to eq('Cool Stuff')
       expect(test_item.description).to eq('supaaa cool')
       expect(test_item.unit_price).to eq(0.1357e2)
       expect(test_item.merchant_id).to eq(12336050)
       expect(test_item.created_at.year).to eq(2016)
       expect(test_item.updated_at.year).to eq(2021)
+
+      test_item = ir.find_by_id(333333333)
+      ir.update(333333333, attributes)
```
[the remark this answers is not in the document]
Comment (Sidarth20) [reply spec/item_repository_spec.rb:218]: it's fine for id number

File: spec/item_repository_spec.rb [hunk 5]
```diff
@@ -177,12 +206,18 @@
...
       ir.update(263567292, attributes)
+
       expect(test_item.name).to eq('Cool Stuff')
       expect(test_item.description).to eq('supaaa cool')
       expect(test_item.unit_price).to eq(0.1357e2)
       expect(test_item.merchant_id).to eq(12336050)
       expect(test_item.created_at.year).to eq(2016)
       expect(test_item.updated_at.year).to eq(2021)
+
+      test_item = ir.find_by_id(333333333)
```
[the remark this answers is not in the document]
Comment (Sidarth20) [reply spec/item_repository_spec.rb:217]: it's fine for id number

File: spec/item_repository_spec.rb [hunk 6]
```diff
@@ -287,4 +325,38 @@
...
+  describe '#items_created_in_month' do
+    it 'creates a hash of number of items created in given month ' do
+      se = SalesEngine.from_csv(
+          items: './spec/truncated_data/items_truncated.csv',
+          merchants: './spec/truncated_data/merchants_truncated.csv',
+          invoices: './spec/truncated_data/invoices_truncated.csv',
+          customers: './spec/truncated_data/customers_truncated.csv',
+          invoice_items: './spec/truncated_data/invoice_items_truncated.csv',
+          transactions: './spec/truncated_data/transactions_truncated.csv'
+                                )
```
[the remark this answers is not in the document]
Comment (Sidarth20) [reply spec/item_repository_spec.rb]: fixed.

File: spec/item_repository_spec.rb [hunk 7]
```diff
@@ -287,4 +325,38 @@
...
+
+      expect(ir.merchants_with_only_one_item.count).to eq(3)
+      expect(ir.merchants_with_only_one_item[1].name).to eq('MiniatureBikez')
+    end
+  end
+
+  describe '#items_created_in_month' do
+    it 'creates a hash of number of items created in given month ' do
+      se = SalesEngine.from_csv(
+          items: './spec/truncated_data/items_truncated.csv',
```
Comment (Sidarth20) [inline spec/item_repository_spec.rb]: Updated

File: spec/item_repository_spec.rb [hunk 8]
```diff
@@ -287,4 +325,38 @@
...
+  describe '#merchants_with_only_one_item' do
+    it 'finds merchants with only 1 item and returns item object' do
+      se = SalesEngine.from_csv(
+          items: './spec/truncated_data/items_truncated.csv',
+          merchants: './spec/truncated_data/merchants_truncated.csv',
+          invoices: './spec/truncated_data/invoices_truncated.csv',
+          customers: './spec/truncated_data/customers_truncated.csv',
+          invoice_items: './spec/truncated_data/invoice_items_truncated.csv',
+          transactions: './spec/truncated_data/transactions_truncated.csv'
+                                )
```
[the remark this answers is not in the document]
Comment (Sidarth20) [reply spec/item_repository_spec.rb]: Updated

File: spec/item_repository_spec.rb [hunk 9]
```diff
@@ -287,4 +325,38 @@
       expect(ir.golden_items).to eq([])
     end
   end
+
+  describe '#merchants_with_only_one_item' do
+    it 'finds merchants with only 1 item and returns item object' do
+      se = SalesEngine.from_csv(
+          items: './spec/truncated_data/items_truncated.csv',
```
Comment (Sidarth20) [inline spec/item_repository_spec.rb]: Updated

File: spec/item_repository_spec.rb [hunk 10]
```diff
@@ -230,14 +267,14 @@
...
-      se = SalesEngine.from_csv({
+      se = SalesEngine.from_csv(
           items: './spec/truncated_data/items_truncated.csv',
           merchants: './spec/truncated_data/merchants_truncated.csv',
           invoices: './spec/truncated_data/invoices_truncated.csv',
           customers: './spec/truncated_data/customers_truncated.csv',
           invoice_items: './spec/truncated_data/invoice_items_truncated.csv',
           transactions: './spec/truncated_data/transactions_truncated.csv'
-                                })
+                                )
```
[the remark this answers is not in the document]
Comment (Sidarth20) [reply spec/item_repository_spec.rb]: Updated.

**Comment by netia1128** [comment]
Fantastic code Sid!

**Pull request merged**